adcircpy/server/server_config.py

```diff
@@ -0,0 +1,125 @@
+from abc import ABC
+from datetime import timedelta
+import os
+import pathlib
+from tempfile import TemporaryDirectory
+
+PADCIRC_DRIVER_SCRIPT_FILENAME = pathlib.Path(os.path.dirname(__file__)).resolve() / 'padcirc_driver.sh'
+
+argument_sbatch_translations = {
+    'run_directory': 'D',
+    'run_name'     : 'J',
+    'account'      : 'A',
+    'log_filename' : 'output',
+    'slurm_ntasks' : 'n',
+    'duration'     : 'time',
+    'partition'    : 'partition'
+}
+
+
+class ServerConfig(ABC):
+    def __init__(self, script: str):
+        self.script = script
+
+    def run(self):
```

Comment:
I'm a little concerned that this could "conflict" in some sense with the AdcircRun.run() method. How would the AdcircRun class know to execute this method or it's own?

Reply:
TODO for future: resolve all these entry points into running adcirc (shell script, `ServerConfig.run()`, `AdcircRun.run()`, etc.)

Reply:
Let's start documenting this so we can work on it.

If AdcircRun is instantiated with:
  - server_config=None then AdcircRun.run() uses the Python implementation.
  - server_config=ServerConfig() instance, then the AdcircRun.run() method uses ssh (should be implemented on ServerConfig.run())
  - server_config=SlurmConfig() then AdcircRun.run() should deploy the slurm job to a server (can be remote or local)

We would need to go into details about this on a call.